core/lib/src/request/from_request.rs

```diff
@@ -483,6 +489,18 @@ impl<'r> FromRequest<'r> for IpAddr {
     }
 }
 
+#[crate::async_trait]
+impl<'r> FromRequest<'r> for ProxyProto<'r> {
+    type Error = std::convert::Infallible;
+
+    async fn from_request(request: &'r Request<'_>) -> Outcome<Self, Self::Error> {
+        match request.proxy_proto() {
+            Some(proto) => Success(proto),
+            None => Forward(Status::NotFound),
```

Comment:
```suggestion
            None => Forward(Status::InternalServerError),
```

Reply:
Let me get this right; if one adds `ProxyProto` as a request guard to a particular route, and no such header is set (and no other matching route exists), this change would make it return an internal server error?

Reply:
That's right. The idea is that if a route requests a `ProxyProto` directly, then that implies that they expect it. If it's not present, then that's a logic error, which we signal via the `500`. If instead the user is unsure if the value is present, they can instead use an `Option<ProxyProto>`, for example, which is infallible.